flowtracker/src/main/scala/FlowTracker.scala

```diff
@@ -131,29 +132,52 @@ class FlowTracker(val flow: Flow[_],
   def this(flow: Flow[_], runCompleted: AtomicBoolean) = this(flow, runCompleted, "", false)
 
   override def run(): Unit = {
-    try {
+    Try {
       initializeTrackedJobState
       registerShutdownHook
+    }.recover {
+      case t: Throwable =>
+        LOG.warn("Failed to initialize FlowTracker.  This is NOT a fatal error. " +
+          "The job will run as normal, but it will not be tracked by Sahale.", t)
+        runCompleted.set(true)
+    }
+
+    val maxFailures = 10
+    var numFailures = 0
+    while(!runCompleted.get) {
+      Try {
+          updateSteps
+          updateFlow
+          updateAggregates
+          if (!disableProgressBar) {
+            logFlowStatus
+          }
+        }.recover {
+          case t: Throwable if 1 + numFailures < maxFailures=>
+            LOG.warn("FlowTracker has thrown an exception because an update iteration failed. " +
```

Comment:
Can we maybe use a heredoc with | + `stripMargin`? Maybe not really worth it. Also totally stylistic.

Reply:
Hey sure @nixsticks that does clean it up nicely.  I just pushed a commit that does this.